jinahub/rankers/DPRReaderRanker/dpr_reader.py

```diff
@@ -83,53 +80,46 @@ def rank(
         Extracts answers from existing matches, (re)ranks them, and replaces the current
         matches with extracted answers.
 
-        For each match ``num_spans_per_match`` of answers will be extracted, which
+        For each match `num_spans_per_match` of answers will be extracted, which
         means that the new matches of the document will have a length of previous
-        number of matches times ``num_spans_per_match``.
+        number of matches times `num_spans_per_match`.
 
-        The new matches will be have a score called ``relevance_score`` saved under
-        their scores. They will also have a tag ``span_score``, which refers to their
+        The new matches will be have a score called `relevance_score` saved under
+        their scores. They will also have a tag `span_score`, which refers to their
         span score, which is used to rank answers that come from the same match.
 
-        If you specified ``title_tag_key`` at initialization, the tag ``title`` will
+        If you specified `title_tag_key` at initialization, the tag `title` will
         also be added to the new matches, and will equal the title of the match from
         which they were extracted.
 
         :param docs: Documents whose matches to re-rank (specifically, the matches of
             the documents on the traversal paths will be re-ranked). The document's
-            ``text`` attribute is taken as the question, and the ``text`` attribute
-            of the matches as the context. If you specified ``title_tag_key`` at
+            `text` attribute is taken as the question, and the `text` attribute
+            of the matches as the context. If you specified `title_tag_key` at
             initialization, the matches must also have a title (under this tag).
-        :param parameters: dictionary to define the ``traversal_path`` and the
-            ``batch_size``. For example::
-            parameters={'traversal_paths': ['r'], 'batch_size': 10}
+        :param parameters: dictionary to define the `traversal_paths` and the
+            `batch_size`. For example
+            `parameters={'traversal_paths': ['r'], 'batch_size': 10}`
         """
 
         if not docs:
             return None
 
-        for doc in docs.traverse_flat(
-            parameters.get('traversal_paths', self.traversal_paths)
+        for doc_arr in docs.batch(
+            batch_size=1,
```

Comment:
why batch size is hard coded here?

Reply:
I see that this outer loop has caused some confusion. Let me explain what is happening here:
- first, we loop over documents in `docs` (doing this one by one)
- then, we loop over matches of a document in the outer loop (doing this in batches). This is the processing step where we actually use batching (to feed a batch of text to the model).

The outer loop can not really use batching, as we need to "attach" the new matches back to the original document. The use of `batch` there was just a convenience. I have now replaced it with `traverse_flat`, to avoid any misunderstanding. Please re-check @nan-wang @bwanglzu